contracts/state/State.sol

```diff
@@ -145,7 +145,9 @@ contract State is Ownable2StepUpgradeable, IState {
         bytes calldata methodParams
     ) public {
         if (methodId == 1) {
-            uint256 calcId = GenesisUtils.calcIdFromEthAddress(getDefaultIdType(), msg.sender);
+            bytes2 idType = GenesisUtils.getIdType(id);
```

Comment:
you also need to check if it is supported in the contract

Reply:
what do you mean?

Reply:
you need not only extract type from id, but also check that it is supported in the contract, so we can avoid state transition for IDs which we don't support.

Reply:
Still don't get it... Is this line is not enough `require(_stateData.isIdTypeSupported[idType], "id type is not registered")`???